SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -663,46 +664,47 @@ public class SignalAttachment: NSObject {
         guard let image = image else {
             let dataSource = DataSourceValue.emptyDataSource()
             dataSource.sourceFilename = filename
-            let attachment = SignalAttachment(dataSource:dataSource, dataUTI: dataUTI)
+            let attachment = SignalAttachment(dataSource: dataSource, dataUTI: dataUTI)
             attachment.error = .missingData
             return attachment
         }
 
         // Make a placeholder attachment on which to hang errors if necessary.
         let dataSource = DataSourceValue.emptyDataSource()
         dataSource.sourceFilename = filename
-        let attachment = SignalAttachment(dataSource : dataSource, dataUTI: dataUTI)
+        let attachment = SignalAttachment(dataSource: dataSource, dataUTI: dataUTI)
         attachment.cachedImage = image
 
         Logger.verbose("\(TAG) Writing \(attachment.mimeType) as image/jpeg")
-        return compressImageAsJPEG(image : image, attachment : attachment, filename:filename, imageQuality:imageQuality)
+        return compressImageAsJPEG(image: image, attachment: attachment, filename: filename, imageQuality: imageQuality)
     }
 
     private class func compressImageAsJPEG(image: UIImage, attachment: SignalAttachment, filename: String?, imageQuality: TSImageQuality) -> SignalAttachment {
         assert(attachment.error == nil)
 
         if imageQuality == .original &&
-            attachment.dataLength < kMaxFileSizeGeneric {
+            attachment.dataLength < kMaxFileSizeGeneric &&
+            outputImageUTISet.contains(attachment.dataUTI) {
```

Comment:
Second fix, always convert HEIF/HEIC to jpg, even if in "original quality" mode.